sonar-db/src/main/java/org/sonar/db/organization/OrganizationDao.java

```diff
@@ -61,6 +62,41 @@ public void insert(DbSession dbSession, OrganizationDto organization) {
     return Optional.ofNullable(getMapper(dbSession).selectByKey(key));
   }
 
+  public List<OrganizationDto> selectByUuids(DbSession dbSession, Set<String> organizationUuids) {
+    if (organizationUuids.size() == 1) {
+      return Collections.singletonList(getMapper(dbSession).selectByUuid(organizationUuids.iterator().next()));
+    }
+    return executeLargeInputs(organizationUuids, getMapper(dbSession)::selectByUuids);
+  }
+
+  /**
+   * Retrieve the default template of the specified organization if:
+   * <ol>
+   *   <li>the specified organization exists</li>
+   *   <li>the global default permission template is defined</li>
+   * </ol>
+   *
+   * @throws NullPointerException whenever the returned DefaultTemplate would contain a {@code null}
+   *         {@link DefaultTemplates#project project}, to prevent returning inconsistent data,
+   */
+  public Optional<DefaultTemplates> getDefaultTemplates(DbSession dbSession, String organizationUuid) {
+    checkUuid(organizationUuid);
+    Optional<DefaultTemplates> res = Optional.ofNullable(getMapper(dbSession).selectDefaultTemplatesByUuid(organizationUuid));
+    res.ifPresent(dto -> {
+      checkNotNull(dto.getProject(), "Inconsistent data in table Organizations for uuid '%s'. " +
+        "Column default_perm_template_project can't be null when column default_perm_template_view is non null",
```

Comment:
This contradicts `DefaultTemplates.project()` which can't return null, even if view() is not null.